### ISSUE TITLE
"system error message could not be fetched" error after multiple executions of "spawn_process"

### ISSUE THREAD

**Issue by TimtheG** [lbartnik/subprocess#55]
Hello Mr. Bartnik,

first of all thank you for the package, it is really helpful!
However, i encoutered a problem which i cannot fix by myself. After several executions of `spawn_subprocess` i get the following error message:

```
Error in process_kill(processHandle) : 
  system kill() failed: system error message could not be fetched
```

a minimal working example (making use of your example code) looks like this:

```
library(subprocess)
is_windows <- function () (tolower(.Platform$OS.type) == "windows")
R_binary <- function () {
  R_exe <- ifelse (is_windows(), "R.exe", "R")
  return(file.path(R.home("bin"), R_exe))
}

for(iProcess in 1:500){
  processHandle = spawn_process(R_binary())
  print(iProcess)
  process_kill(processHandle)
}
```
In the actual application, the code ran for about half an hour and spawned subprocesses that exited the spawned R session istself (without `process_kill`). The error message was slightly different then and occured in the next execution of "spawn_subprocess" ("could not create pipe"). The error occured after about 100-300 spawned processes, depending on the code that was executed within the subprocess. I used Ubuntu 16.04.3 LTS with R Version 3.4.3. Running it on a windows machine did not create any errors.

I would be very grateful if you could provide a solution to this problem!
Tim

**Comment by ws171913** [comment]
Hello,

I managed to reproduce the issue. I suppose problems are caused by session information beeing stored by R process. Probably R it blocks file to store environment in and subsequent process cannot access it. If you try to read stderr of process you will see that some of processes out of 500 complain about lack of --no-save or --vanilla argument. 

It's process starting/termination race. On linux process management is completely different than on windows so windows was not issue.

By the way I stamped onto diffrent problem on linux. Reading from pipes from process was realised with select, which has limitation to pipe handler value (it cannot be over 1024). As you create so much processes each of them creates at least 3 double pipes (for read & white) so with select after creation of arround 170 subprocesses reading subprocess output caused crash. I have changed the mechanizm to use poll instead. It works now.

All my changes are pending for review and suppose will be available soon.

**Comment by lbartnik** [comment]
With a minimal change in the native code I've been able to retrieve the `errno` value set when `kill()` fails. It has value `3`, that is, `ESRCH`. Which validates what @ws171913 wrote about killing the child process before it is able to actually start.

I made a tiny little change to your code and was able to get past this first problem. Here's the new loop:

```r
ph <- lapply(seq(500), function(i) {
  print(i)
  spawn_process(R_binary(), "--slave")
})
pk <- lapply(ph, process_kill)
```

Now I get to 339 child processes and fail with `errno = 24` (`EMFILE` Too many open files). This can be remediated by calling `ulimit -S -n 4000` in bash. After this change of limit of open files, I reach 500 child processes running at the same time.

That being said, unless you disagree and can provide a compelling reason, I'm not going to include a "control pipe" to further the parent-child synchronization moment.
